src/routes/userRoutes.js

```diff
@@ -81,6 +81,31 @@ routes.deleteUser = async (req, res) => {
       res.status(500).json({error: {message: 'Deleting user failed. Please try again'}})
   }
 }
+  routes.updateDBPassword = async(req,res) =>{
+    if(req.params.id===null ||!req.body.password ){
+      logger.info('invalid input')
+      return res.status(400).json({error:{message:'invalid input of userid and password'}})
+    }
+    const {Accounts} = db.getModels()
+    const userAccount = await Accounts.findOne({
+        where:{
+          id:req.params.id
+        }
+    })
+    if(!userAccount){
+        logger.info(`account does not exist`)
+        res.status(400).json({error:{message:"account does not exist"}})
```

Comment:
`status(404)` for not found

Reply:
404 is page not found. Here is the account does not exist. 400 is bad request error; which we use here to indicate the request body id don’t match with any account in our database.

Reply:
I thought so too but I was convinced by these people


https://stackoverflow.com/questions/5604816/whats-the-most-appropriate-http-status-code-for-an-item-not-found-error-page

Reply:
In that case, shall we replace all the 400 we used in the code to 404 (say invalid input and account not found)? 

10.4.1 400 Bad Request

The request could not be understood by the server due to malformed syntax. The client SHOULD NOT repeat the request without modifications.


10.4.5 404 Not Found

The server has not found anything matching the Request-URI. No indication is given of whether the condition is temporary or permanent. The 410 (Gone) status code SHOULD be used if the server knows, through some internally configurable mechanism, that an old resource is permanently unavailable and has no forwarding address. This status code is commonly used when the server does not wish to reveal exactly why the request has been refused, or when no other response is applicable.

https://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html